src/industry_gui.cpp

```diff
@@ -1364,7 +1374,7 @@ class IndustryDirectoryWindow : public Window {
 	{
 		assert(id < lengthof(i->produced_cargo));
 
-		if (i->produced_cargo[id] == CT_INVALID) return 101;
+		if (i->produced_cargo[id] == CT_INVALID) return 0;
```

Comment:
I've no idea what 101 was doing, but the docstring says -1 if industry doesn't use this cargo slot. Either this return value is now (or still) wrong, or the doc comment needs updating

Reply:
Looking at intended usage, you're now checking outside the function. Perhaps a rename to remove "IfValid" and convert this if into an assert instead?

Reply:
I made it match the docstring, -1, and work around with the returned value in the parent call.